autometrics/src/lib.rs

```diff
@@ -229,5 +231,19 @@ pub mod __private {
         LocalKey { inner: CALLER_KEY }
     };
 
-    pub use spez::spez;
+    /// Load the service name from one of the following runtime environment variables:
+    /// - `AUTOMETRICS_SERVICE_NAME`
+    /// - `OTEL_SERVICE_NAME`
+    /// Or, fall back to the name of the cargo package defined in the `Cargo.toml` file.
+    pub fn service_name(cargo_pkg_name: &'static str) -> &'static str {
+        // Note that we are using a OnceCell here because we want the label value to
+        // be available as a &'static str. This allows us to load the value from the
+        // environment variables once at startup while still accessing it as a &'static str
+        static SERVICE_NAME: OnceCell<String> = OnceCell::new();
+        SERVICE_NAME.get_or_init(|| {
+            std::env::var("AUTOMETRICS_SERVICE_NAME")
+                .or_else(|_| std::env::var("OTEL_SERVICE_NAME"))
+                .unwrap_or(cargo_pkg_name.to_string())
```

Comment:
for owned variants `.unwrap_or_else` should be preferred (even though in this case it doesnt really make a difference as this gets called exactly once)

Reply:
👍 I'm surprised that clippy didn't catch that